### PR TITLE
Add placeholder to search input in Select component

### PR DESCRIPTION
**Pull request by dmandrade**
Add option to set a placeholder in the search field providing a better ux when the select already has preloaded data.

image

image

### CODE REVIEW

File: packages/forms/resources/js/components/select.js [hunk 1]
```diff
@@ -58,6 +59,7 @@ export default (Alpine) => {
                         renderChoiceLimit: optionsLimit,
                         searchFields: ['label'],
                         searchResultLimit: optionsLimit,
+                        searchPlaceholderValue: searchPlaceholder,
```
Comment (zepfietje) [inline packages/forms/resources/js/components/select.js]:
```suggestion
                        searchPlaceholderValue: searchPlaceholder,
                        searchResultLimit: optionsLimit,
```

File: packages/forms/resources/js/components/select.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export default (Alpine) => {
             searchDebounce,
             searchingMessage,
             searchPrompt,
+            searchPlaceholder,
```
Comment (zepfietje) [inline packages/forms/resources/js/components/select.js]:
```suggestion
            searchPlaceholder,
            searchPrompt,
```

File: packages/forms/resources/lang/en/components.php [hunk 3]
```diff
@@ -250,6 +250,8 @@
 
         'search_prompt' => 'Start typing to search...',
 
+        'search_placeholder' => 'Search',
+
```
Comment (zepfietje) [inline packages/forms/resources/lang/en/components.php]:
```suggestion

        'search_placeholder' => 'Search',

        'search_prompt' => 'Start typing to search...',
```

File: packages/forms/resources/lang/en/components.php [hunk 4]
```diff
@@ -248,8 +248,9 @@
 
         'searching_message' => 'Searching...',
 
-        'search_prompt' => 'Start typing to search...',
+        'search_placeholder' => 'Search',
 
+        'search_prompt' => 'Start typing to search...',
```
Comment (zepfietje) [inline packages/forms/resources/lang/en/components.php]:
```suggestion
        'search_prompt' => 'Start typing to search...',

```

File: packages/forms/resources/lang/pt_BR/components.php [hunk 5]
```diff
@@ -249,6 +249,8 @@
         'searching_message' => 'Buscando...',
 
         'search_prompt' => 'Comece a digitar para pesquisar...',
+
+        'search_placeholder' => 'Buscar',
```
Comment (zepfietje) [inline packages/forms/resources/lang/pt_BR/components.php]:
```suggestion
        'search_placeholder' => 'Buscar',

        'search_prompt' => 'Comece a digitar para pesquisar...',

```

File: packages/forms/resources/views/components/select.blade.php [hunk 6]
```diff
@@ -99,6 +99,7 @@
                         searchDebounce: @js($getSearchDebounce()),
                         searchingMessage: @js($getSearchingMessage()),
                         searchPrompt: @js($getSearchPrompt()),
+                        searchPlaceholder: @js($getSearchPlaceholder()),
```
Comment (zepfietje) [inline packages/forms/resources/views/components/select.blade.php]:
```suggestion
                        searchPlaceholder: @js($getSearchPlaceholder()),
                        searchPrompt: @js($getSearchPrompt()),
```

File: packages/forms/src/Components/Concerns/CanBeSearchable.php [hunk 7]
```diff
@@ -17,6 +17,8 @@ trait CanBeSearchable
 
     protected string | Htmlable | Closure | null $searchPrompt = null;
 
+    protected string | Closure | null $searchPlaceholder = null;
```
Comment (zepfietje) [inline packages/forms/src/Components/Concerns/CanBeSearchable.php]:
```suggestion
    protected string | Closure | null $searchPlaceholder = null;

    protected string | Htmlable | Closure | null $searchPrompt = null;
```

File: packages/forms/src/Components/Concerns/CanBeSearchable.php [hunk 8]
```diff
@@ -52,6 +54,13 @@ public function searchPrompt(string | Htmlable | Closure | null $message): stati
         return $this;
     }
 
+    public function searchPlaceholder(string | Closure | null $message): static
+    {
+        $this->searchPlaceholder = $message;
+
+        return $this;
+    }
+
```
Comment (zepfietje) [inline packages/forms/src/Components/Concerns/CanBeSearchable.php]: Could you move this above `searchPrompt` to ensure alphabetical sort, @dmandrade?

File: packages/forms/src/Components/Concerns/CanBeSearchable.php [hunk 9]
```diff
@@ -62,6 +71,11 @@ public function getSearchPrompt(): string | Htmlable
         return $this->evaluate($this->searchPrompt) ?? __('forms::components.select.search_prompt');
     }
 
+    public function getSearchPlaceholder(): string | Htmlable
+    {
+        return $this->evaluate($this->searchPlaceholder) ?? __('forms::components.select.search_placeholder');
+    }
+
```
Comment (zepfietje) [inline packages/forms/src/Components/Concerns/CanBeSearchable.php]: Move this too.

**Review by zepfietje — CHANGES_REQUESTED** [review]
Mark as ready for review when done.

**Comment by danharrin** [comment]
Why can't we use the existing `searchPrompt()`?

**Comment by dmandrade** [comment]
> Why can't we use the existing `searchPrompt()`?

I thought about it but I believe that choicesjs will always display a message under the search field when there are no preloaded items, so I thought it best to add one more configuration option.

**Comment by zepfietje** [comment]
Can we hide the prompt? Then just displaying the placeholder instead should be fine maybe?

**Comment by dmandrade** [comment]
I can do that. 
do you think it would be a better ux just display the input with placeholder inside?

**Comment by danharrin** [comment]
We shouldnt need to add any more methods to the class to do this

**Comment by dmandrade** [comment]
New methods have been removed and the class is as it was before.
just passing the searchPrompt as placeholder and hiding the "has-no-choices" element when the field is a single selection.

**Comment by danharrin** [comment]
Thanks

**Pull request merged**